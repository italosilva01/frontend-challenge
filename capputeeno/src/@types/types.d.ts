export interface Product {
  image_url: string;
  name: string;
  sales: number;
  price_in_cents: string;
  created_at: string;
}

export type TabType = 'all' | 't-shirts' | 'mugs';
