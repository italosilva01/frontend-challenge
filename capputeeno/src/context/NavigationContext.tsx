import React, { ReactNode, createContext, useContext, useState } from 'react';
import { TabType } from '../@types/types';

const NavigationContext = createContext<NavigationContextData>(
  [] as unknown as NavigationContextData
);

interface NavigationContextData {
  page: number;
  categoryProduct: TabType;
  changePage: (val: number) => void;
  changeCategory: (val: TabType) => void;
}

interface NavigationProvider {
  children: ReactNode;
}

export const NavigationProvider = ({ children }: NavigationProvider) => {
  const [page, setPage] = useState(1);
  const [categoryProduct, setCategoryProduct] = useState<TabType>('all');

  const changePage = (val: number) => {
    setPage(val);
  };
  const changeCategory = (val: TabType) => {
    setCategoryProduct(val);
  };

  return (
    <NavigationContext.Provider
      value={{
        page,
        categoryProduct,
        changePage,
        changeCategory,
      }}
    >
      {children}
    </NavigationContext.Provider>
  );
};

export function useNavigation(): NavigationContextData {
  const context = useContext(NavigationContext);

  return context;
}
