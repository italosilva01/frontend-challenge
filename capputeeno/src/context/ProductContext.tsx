import React, { ReactNode, useState, useContext, createContext } from 'react';
import { Product } from '../@types/types';

const ProductContext = createContext<ProductContextData>([]);

interface ProductProviderProps {
  children: ReactNode;
}

interface ProductContextData {
  products: Product[];
  addProducts: (newProducts: Product[]) => void;
}

export const ProductProvider = ({ children }: ProductProviderProps) => {
  const [products, setProducts] = useState<Product[]>([] as Product[]);

  const addProducts = (newProducts: Product[]) => {
    console.log(newProducts);
    setProducts(newProducts);
  };

  return (
    <ProductContext.Provider
      value={{
        addProducts,
        products,
      }}
    >
      {children}
    </ProductContext.Provider>
  );
};

export function useProduct(): ProductContextData {
  const context = useContext(ProductContext);

  return context;
}
