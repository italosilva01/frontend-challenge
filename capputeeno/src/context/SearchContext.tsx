import React, { ReactNode, useContext, useState } from 'react';

const SearchContext = React.createContext<SearchContextData>(
  [] as unknown as SearchContextData
);

interface SearchContextData {
  changeCurrentPage: (v: number) => void;
  currentPage: number;
}

interface SearchProviderProps {
  children: ReactNode;
}
export const SearchProvider = ({ children }: SearchProviderProps) => {
  const [currentPage, setCurrentPage] = useState<number>(1);
  const changeCurrentPage = (newValue: number) => {
    setCurrentPage(newValue);
  };

  return (
    <SearchContext.Provider
      value={{
        currentPage,
        changeCurrentPage,
      }}
    >
      {children}
    </SearchContext.Provider>
  );
};

export function useSearch(): SearchContextData {
  const context = useContext(SearchContext);

  return context;
}
