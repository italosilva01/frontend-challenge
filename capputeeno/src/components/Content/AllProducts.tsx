import React, { useEffect } from 'react';
import { Box, Grid } from '@mui/material';

import { CardItemMinInfo } from '../CardItemMinInfo';

import { useProduct } from '../../context/ProductContext';
import { EmptySearch } from '../EmptySearch';
import { Product } from '../../@types/types';

interface AllProductsProps {
  initProducts: Product[];
}

export const AllProducts = ({ initProducts }: AllProductsProps) => {
  const { addProducts, products } = useProduct();

  useEffect(() => {
    if (initProducts)
      addProducts(Object.values(initProducts)[0] as unknown as Product[]);
  }, []);

  return (
    <>
      <Box>
        <Grid
          container
          spacing={2}
          columns={{ xs: 4, md: 4 }}
          className=" mx-auto w-[100%]   h-[100%] "
          alignItems="center"
          justifyContent="center"
        >
          {products?.length === 0 ? (
            <EmptySearch />
          ) : (
            products?.map(({ name, image_url, price_in_cents, id }, index) => (
              <Grid item key={index} md>
                <CardItemMinInfo
                  idProduct={id}
                  productPrice={price_in_cents}
                  imageUrl={image_url}
                  productName={name}
                />
              </Grid>
            ))
          )}
        </Grid>
      </Box>
    </>
  );
};
