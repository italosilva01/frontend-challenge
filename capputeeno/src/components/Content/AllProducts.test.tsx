import React from 'react';
import { useProduct } from '../../context/ProductContext';
import { AllProducts } from './AllProducts';
import { render } from '@testing-library/react';
import { Product } from '../../@types/types';
useProduct;

const mockUseProduct = jest.fn();
jest.mock('../../context/ProductContext', () => {
  return {
    useProduct: jest.fn(),
  };
});

const mockProducts = [
  {
    image_url:
      'https://storage.googleapis.com/xesque-dev/challenge-images/camiseta-02.jpg',
    name: 'Camiseta evening',
    price_in_cents: 5859,
    id: '2bfe252f-c2d1-4bcc-b3c3-552e549df9ac',
  },
  {
    image_url:
      'https://storage.googleapis.com/xesque-dev/challenge-images/caneca-02.jpg',
    name: 'Caneca Decaf! P&Co',
    price_in_cents: 4673,
    id: '3ccbda00-5694-4289-8b08-f1840fe3a89c',
  },
  {
    image_url:
      'https://storage.googleapis.com/xesque-dev/challenge-images/caneca-06.jpg',
    name: 'Caneca de cerâmica rústica',
    price_in_cents: 5859,
    id: '168b613a-c93f-476e-98b4-4aa6ab71d32a',
  },
];
describe('Test the component what should show all Products', () => {
  beforeEach(() => {
    (useProduct as jest.Mock).mockReturnValue({
      products: mockProducts,
      addProducts: jest.fn(),
    });
  });

  test('Should show all products of home page', () => {
    const { container } = render(
      <AllProducts initProducts={[] as Product[]} />
    );

    expect(container).toMatchSnapshot();
  });
});
