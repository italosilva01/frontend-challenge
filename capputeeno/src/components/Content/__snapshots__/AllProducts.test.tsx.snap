// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the component what should show all Products Should show all products of home page 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2  mx-auto w-[100%]   h-[100%]  css-919fip-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-true css-ehu4kr-MuiGrid-root"
      >
        <div
          class=" rounded-lg bg-white w-[16rem] min-h-min cursor-pointer  MuiBox-root css-0"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27256%27%20height=%27300%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              class="border rounded-t-lg"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <div
            class="grid px-3 py-2 gap-y-2 MuiBox-root css-0"
          >
            <p
              class="prose-base font-['Saira'] text-base leading-6 "
              data-testid="product-name"
            >
              Camiseta evening
            </p>
            <div
              class="w-[14.25rem] h-[1px] border border-[DCE2E5] mx-auto"
            />
            <p
              class="font-bold"
              data-testid="product-cost"
            >
              R$ 58,59
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-true css-ehu4kr-MuiGrid-root"
      >
        <div
          class=" rounded-lg bg-white w-[16rem] min-h-min cursor-pointer  MuiBox-root css-0"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27256%27%20height=%27300%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              class="border rounded-t-lg"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <div
            class="grid px-3 py-2 gap-y-2 MuiBox-root css-0"
          >
            <p
              class="prose-base font-['Saira'] text-base leading-6 "
              data-testid="product-name"
            >
              Caneca Decaf! P&Co
            </p>
            <div
              class="w-[14.25rem] h-[1px] border border-[DCE2E5] mx-auto"
            />
            <p
              class="font-bold"
              data-testid="product-cost"
            >
              R$ 46,73
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-true css-ehu4kr-MuiGrid-root"
      >
        <div
          class=" rounded-lg bg-white w-[16rem] min-h-min cursor-pointer  MuiBox-root css-0"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27256%27%20height=%27300%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              class="border rounded-t-lg"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <div
            class="grid px-3 py-2 gap-y-2 MuiBox-root css-0"
          >
            <p
              class="prose-base font-['Saira'] text-base leading-6 "
              data-testid="product-name"
            >
              Caneca de cerâmica rústica
            </p>
            <div
              class="w-[14.25rem] h-[1px] border border-[DCE2E5] mx-auto"
            />
            <p
              class="font-bold"
              data-testid="product-cost"
            >
              R$ 58,59
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
